api4/user.go

```diff
@@ -550,3 +552,65 @@ func getAudits(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 }
+
+func updateMfa(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if !app.SessionHasPermissionToUser(c.Session, c.Params.UserId) {
+		c.SetPermissionError(model.PERMISSION_EDIT_OTHER_USERS)
+		return
+	}
+
+	props := model.StringInterfaceFromJson(r.Body)
+
+	activate, ok := props["activate"].(bool)
+	if !ok {
+		c.SetInvalidParam("activate")
+		return
+	}
+
+	token := ""
+	if activate {
+		token = props["token"].(string)
+		if len(token) == 0 {
+			c.SetInvalidParam("token")
+			return
+		}
+	}
+
+	c.LogAudit("attempt")
+
+	if activate {
+		if err := app.ActivateMfa(c.Params.UserId, token); err != nil {
+			c.Err = err
+			return
+		}
+		c.LogAuditWithUserId(c.Params.UserId, "success - activated")
+	} else {
+		if err := app.DeactivateMfa(c.Params.UserId); err != nil {
+			c.Err = err
+			return
+		}
+		c.LogAuditWithUserId(c.Params.UserId, "success - deactivated")
+	}
+
+	go func() {
+		var user *model.User
+		var err *model.AppError
+		if user, err = app.GetUser(c.Session.UserId); err != nil {
+			l4g.Warn(err.Error())
+			return
+		}
+
+		if err := app.SendMfaChangeEmail(user.Email, activate, user.Locale, c.GetSiteURL()); err != nil {
+			l4g.Error(err.Error())
+		}
+	}()
```

Comment:
Would you be able to create an app layer function in `app/user.go` that will take in the `useId`, `token` and `activate`. It should perform the logic on whether to activate or deactivate MFA and it should contain the go routine that sends the email.

The audits can stay in the api layer